src/Application/Executor.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Symbolica.Abstraction;
+using Symbolica.Application.Collection;
+using Symbolica.Application.Computation;
+using Symbolica.Application.Implementation;
+using Symbolica.Computation;
+using Symbolica.Deserialization;
+using Symbolica.Implementation;
+using Symbolica.Implementation.System;
+using Symbolica.Representation;
+
+namespace Symbolica.Application
+{
+    internal static class Executor
+    {
+        public static async Task<Result> Run(string directory,
+            bool useSymbolicGarbage, bool useSymbolicAddresses, bool useSymbolicContinuations)
+        {
+            var buildImage = Environment.GetEnvironmentVariable("DOCKER_BUILD_IMAGE");
```

Comment:
Let's namespace theses with `SYMBOLICA_` instead of `DOCKER_` as they're our env vars.